daemon/cmd/status.go

```diff
@@ -200,6 +210,82 @@ func (d *Daemon) getKubeProxyReplacementStatus() *models.KubeProxyReplacement {
 	}
 }
 
+func (d *Daemon) getBPFMapStatus() *models.BPFMapStatus {
```

Comment:
We could plausibly have this iterate a structure held in the daemon, where that structure is populated via each map registering themselves, eg by passing the daemon into each map creation function which would receive an interface that represents the 'map manager' (with `RegisterMap(...)` method), then have the core bpf lib code automatically register the maps using their name / sizes at creation time. This would mean that we don't have to update this extra function whenever we add/remove maps, it would be automatically kept up-to-date by the core bpf map lib code.

But up to you whether to explore this now or later, it's not such a big problem that I think we need to solve this.

Reply:
Nice, thanks for the suggestion! I like this approach and will look into implementing it in a follow-up PR.